editor/src/components/common/shared-strategies/convert-to-flex-strategy.ts

```diff
@@ -0,0 +1,192 @@
+      setHugContentForAxis('vertical', path),
+      ...setPropertyOmitNullProp('always', path, PP.create('style', 'padding'), padding),
+      ...childrenPaths.flatMap((child) => [
+        ...nukeAllAbsolutePositioningPropsCommands(child),
+        ...sizeToVisualDimensions(metadata, child),
+        ...convertWidthToFlexGrowOptionally(metadata, child),
+      ]),
+    ]
+  })
+}
+
+function guessMatchingFlexSetup(
+  metadata: ElementInstanceMetadataMap,
+  target: ElementPath,
+  children: Array<ElementPath>,
+): {
+  direction: FlexDirection
+  sortedChildren: Array<CanvasFrameAndTarget>
+  averageGap: number | null
+  padding: string | null
+} {
+  const result = guessLayoutDirection(metadata, target, children)
+
+  if (result.sortedChildren.length === 0) {
+    return { ...result, padding: null }
+  }
+
+  const padding: string | null = guessPadding(
+    result.direction,
+    result.parentRect,
+    result.sortedChildren,
+  )
+
+  return { ...result, padding: padding }
+}
+
+function guessLayoutDirection(
+  metadata: ElementInstanceMetadataMap,
+  target: ElementPath,
+  children: Array<ElementPath>,
+): {
+  direction: FlexDirection
+  sortedChildren: Array<CanvasFrameAndTarget>
+  parentRect: CanvasRectangle
+  averageGap: number | null
+} {
+  const parentRect = MetadataUtils.getFrameInCanvasCoords(target, metadata) ?? zeroCanvasRect
+  const firstGuess: FlexDirection = parentRect.width > parentRect.height ? 'row' : 'column'
+  const firstGuessResult = isThereOverlapInDirection(metadata, children, firstGuess, parentRect)
+  if (firstGuessResult.childrenDontOverlap) {
+    return firstGuessResult
+  }
+  const secondGuess = firstGuess === 'row' ? 'column' : 'row'
+  const secondGuessResult = isThereOverlapInDirection(metadata, children, secondGuess, parentRect)
+  if (secondGuessResult.childrenDontOverlap) {
+    return secondGuessResult
+  }
+
+  // since none of the directions are great, let's fall back to our first guess
+  return firstGuessResult
+}
+
+function guessPadding(
+  direction: FlexDirection,
+  parentRect: CanvasRectangle,
+  sortedChildren: Array<CanvasFrameAndTarget>,
+): string | null {
+  const firstChild = sortedChildren[0]
+  const lastChild = last(sortedChildren)!
+
+  if (direction === 'row') {
+    const paddingLeft = firstChild.frame.x - parentRect.x
+    const paddingRight =
+      parentRect.x + parentRect.width - (lastChild?.frame.x + lastChild?.frame.width)
+    const horizontalPadding = Math.min(paddingLeft, paddingRight)
+    if (horizontalPadding === 0) {
+      return null
+    }
+    return `0 ${horizontalPadding}px`
+  } else {
+    const paddingTop = firstChild.frame.y - parentRect.y
+    const paddingBottom =
+      parentRect.y + parentRect.height - (lastChild?.frame.y + lastChild?.frame.height)
+    const verticalPadding = Math.max(0, Math.min(paddingTop, paddingBottom))
+    if (verticalPadding === 0) {
+      return null
+    }
+    return `${verticalPadding}px 0`
+  }
```

Comment:
I'd argue we don't need to handle these separately, and should instead include both horizontal and vertical padding

Reply:
hah, I'll try